plugins/modules/podman_prune.py

```diff
@@ -0,0 +1,293 @@
+    description:
+      - List of IDs of deleted containers.
+    returned: I(containers) is C(true)
+    type: list
+    elements: str
+    sample: []
+
+# images
+images:
+    description:
+      - List of IDs of deleted images.
+    returned: I(images) is C(true)
+    type: list
+    elements: str
+    sample: []
+
+# networks
+networks:
+    description:
+      - List of IDs of deleted networks.
+    returned: I(networks) is C(true)
+    type: list
+    elements: str
+    sample: []
+
+# volumes
+volumes:
+    description:
+      - List of IDs of deleted volumes.
+    returned: I(volumes) is C(true)
+    type: list
+    elements: str
+    sample: []
+
+# system
+system:
+  description:
+    - List of ID of deleted containers, volumes, images, network and total reclaimed space
+  returned: I(system) is C(true)
+  type: list
+  elements: str
+  sample: []
+"""
+
+
+from ansible.module_utils.basic import AnsibleModule
+
+
+def podmanExec(module, target, filters, executable):
+    changed = False
+    command = [executable, target, 'prune', '--force']
+    if filters != "" and target != "system":
+        command.append("--filter=")
+        command.append(filters)
+    if filters != "" and target == "system":
+        split_filters = filters.strip().split(" ")
+        for filter in split_filters:
+            if filter:
+                command.append(filter)
+    rc, out, err = module.run_command(command)
+    if out:
+        changed = True
+    if rc != 0:
+        module.fail_json(
+            msg="Error executing prune on {target}: {err}".format(target=target, err=err))
+    return changed, out, err
+
+
+def run_module():
+    module_args = dict(
+        container=dict(type='bool', default=False),
+        container_filters=dict(type='dict'),
+        image=dict(type='bool', default=False),
+        image_filters=dict(type='dict'),
+        network=dict(type='bool', default=False),
+        network_filters=dict(type='dict'),
+        volume=dict(type='bool', default=False),
+        volume_filters=dict(type='dict'),
+        system=dict(type='bool', default=False),
+        system_all=dict(type='bool', default=False),
+        system_volumes=dict(type='bool', default=False),
+        executable=dict(type='str', default='podman')
+    )
+
+    module = AnsibleModule(
+        argument_spec=module_args
+    )
+
+    executable = module.get_bin_path(
+        module.params['executable'], required=True)
+
+    if module.params["container"]:
+        target = "container"
+        if not module.params["container_filters"]:
+            filters = ""
+        else:
+            filters = module.params["container_filters"]
+        changed, out, err = podmanExec(module, target, filters, executable)
+        if not out:
+            containers = []
+        else:
+            containers = out.rstrip().split("\n")
+        results = dict(
+            changed=changed,
+            containers=containers,
+            stderr=err
+        )
```

Comment:
I see this pattern repeats every time, can we maybe to optimize a little bit? Like:
```python
for target, filters in (("container", "container_filters"), ("network", "network_filters"), ("image", "image_filters"), ("volume", ("volume_filters")):
    if module.params[target]:
        changed, out, err = podmanExec(module, target, module.params[filters], executable)